# HG changeset patch
# User Ben Abbott <bpabbott@mac.com>
# Date 1224512944 14400
#      Mon Oct 20 10:29:04 2008 -0400
# Node ID 041bb0c64f07eeaf3bef27fda18c0ad01980c72e
# Parent  6ef647359f076a1864bf22d0ddb35d5b07f923fc
orient.m,print.f: validate figure handle

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,13 +1,15 @@
 2008-10-20  Ben Abbott  <bpabbott@mac.com>
 
 	* plot/__stem__.m: Respect new ordering of children when setting
 	baseline.
 
+	* plot/orient.m, plot/print.m: Properly validate figure handle.
+
 2008-10-20  David Bateman  <dbateman@free.fr>
 
 	* plot/plotyy.m: Test that an axes handle actually is one before
 	setting it.
 
 2008-10-17  David Bateman  <dbateman@free.fr>
 
 	* plot/__plt_get_axis_arg__.m: Exclude non-numeric and root figure
diff --git a/scripts/plot/orient.m b/scripts/plot/orient.m
--- a/scripts/plot/orient.m
+++ b/scripts/plot/orient.m
@@ -27,17 +27,17 @@
 ## Adapted-By: jwe
 
 ## PKG_ADD: mark_as_command orient
 
 function retval = orient (varargin)
 
   nargs = nargin;
 
-  if (nargs > 0 && numel (varargin{1}) == 1 && ishandle (varargin{1}))
+  if (nargs > 0 && numel (varargin{1]) == 1 && isfigure (varargin{1}))
     cf = varargin{1};
     varargin(1) = [];
     nargs--;
   else
     cf = gcf ();
   endif
 
   if (nargs == 0)
diff --git a/scripts/plot/print.m b/scripts/plot/print.m
--- a/scripts/plot/print.m
+++ b/scripts/plot/print.m
@@ -185,17 +185,17 @@ function print (varargin)
 	  endif
         elseif (length (arg) > 2 && arg(1:2) == "-S")
 	  size = arg(3:length(arg));
         elseif (length (arg) >= 1 && arg(1) == "-")
 	  error ("print: unknown option `%s'", arg);
 	elseif (length (arg) > 0)
 	  name = arg;
         endif
-      elseif (ishandle (arg))
+      elseif (isfigure (arg))
         figure (arg);
       else
         error ("print: expecting inputs to be character string options or a figure handle");
       endif
     endfor
 
     doprint = isempty (name);
     if (doprint)
@@ -434,14 +434,14 @@ function print (varargin)
     ## FIXME -- This looks like a dirty, Unix-specific hack.
     ## DAS
     if (doprint)
       system (sprintf ("lpr %s '%s'", printer, printname));
       unlink (printname);
     endif
 
   unwind_protect_cleanup
-    if (ishandle (old_fig))
+    if (isfigure (old_fig))
       figure (old_fig)
     endif
   end_unwind_protect
 
 endfunction
